packages/user-menu/src/UserMenu.js

```diff
@@ -59,51 +71,52 @@ class UserMenu extends Component<Props> {
         alignItems: 'center',
       }}>
         <Avatar
-          image={this.props.avatarUrl}
-          icon={this.props.avatarUrl ? null : 'user'}
+          image={avatarUrl}
+          icon={avatarUrl ? null : 'user'}
           alt="avatar"
           width="48"
           height="48" />
-        <h5 className="u-margin-top-xs u-margin-bottom">
-          {`${this.props.firstName} ${this.props.lastName}`}
-        </h5>
+        <p className="u-margin-top-xs u-margin-bottom h5">
+          {`${firstName} ${lastName}`}
+        </p>
       </div>
     )
   }
 
   renderLoggedIn() {
+    const { children, flyoutSize, logoutUrl } = this.props;
     return (
-      <div style={{ textAlign: 'right'}}>
-        <Flyout
-          trigger={this.renderLoggedInButton()}
-          direction="right"
-          hasPadding={true}
-          >
-          <div>
-            {this.renderProfile()}
-            {this.props.children}
-            <Button
-              onClick={() => window.location.href=this.props.logoutUrl}
-              block
-              type="danger"
-              iconLeft="power-off">
-              Afmelden
-            </Button>
-          </div>
-        </Flyout>
-      </div>
-    )
+      <Flyout
+        trigger={this.renderLoggedInButton()}
+        direction="right"
+        hasPadding={false}
+        size={flyoutSize}
+        >
+        <div>
+          {this.renderProfile()}
+          {children}
+          <Button
+            onClick={() => window.location.href=logoutUrl}
+            block
+            type="danger"
+            iconLeft="power-off">
+            Afmelden
+          </Button>
+        </div>
+      </Flyout>
+    );
   }
 
   renderLoggedOut() {
-    return (
-      <div style={{ textAlign: 'right' }}>
-        <Button iconLeft="user">Aanmelden</Button>
-      </div>);
+    return (<Button iconLeft="user">Aanmelden</Button>);
   }
 
   render() {
-    return (this.props.loggedIn ? this.renderLoggedIn() : this.renderLoggedOut());
+    return (
+      <div style={{ textAlign: this.props.direction }}>
```

Comment:
This is the only piece code where I wonder if it needs to be part of the component (open question).